### ISSUE TITLE
PDP Product Image does not update on swatch selection if not all options are selected

### ISSUE THREAD

**Issue by kevin-carroll-cl** [magento/magento2#8929]
On the Product Detail Page, if more than one type of option is available for the product (example: size and color), if only color is selected the product image does not update.

### Preconditions

1. Magento Community 2.1.5
2. PHP 7.0
3. Configurable Product with color image swatch and size text-swatch where color selection would trigger an update to the product image.

### Steps to reproduce

1. Navigate to the Frontend Product Detail Page of the configurable product from Preconditions.
2. Select only color option.

### Expected result

1. The Product Image should update to the product image corresponding to the selected color swatch.

### Actual result

1. The Product Image only updates if all option types are selected.

Initial Page Load: ![](http://files.classyllama.com/4c5bde5d/20170317r03rp.png)

Only Color Swatch selected and JS error in console: ![](http://files.classyllama.com/4c5bde5d/201703176x58d.png)

Size and Color Selected (also expected result of selecting color): ![](http://files.classyllama.com/4c5bde5d/2017031736f5v.png)

This issue appears to be caused by accessing a property on a null object in `magento/module-swatches/view/frontend/web/js/swatch-renderer.js` on line 714:
```
if (result.oldPrice.amount !== result.finalPrice.amount) {
    $(this.options.slyOldPriceSelector).show();
} else {
    $(this.options.slyOldPriceSelector).hide();
}
```
Issue does not occur if this code is changed to:
```
if (result && result.oldPrice && result.finalPrice && result.oldPrice.amount !== result.finalPrice.amount) {
    $(this.options.slyOldPriceSelector).show();
} else {
    $(this.options.slyOldPriceSelector).hide();
}
```
The above code change is likely not the correct resolution to this issue, it is included more as an illustration of where the breaking occurs.  If all product option types are selected, this result object exists as expected and normal/expected behaviour resumes.

**Comment by drpit** [comment]
I have the same problem, but your solution does not seem to work for me

**Comment by VasylShvorak** [comment]
@kevin-carroll-cl  thank you for your report.
Is this issue still actual?
We were not able to reproduce it on the ver. 2.1.7

**Comment by dneligan** [comment]
The issue happens to me on 2.1.7

**Comment by drpit** [comment]
I had the same issue but disappeared on after upgrading to 2.1.7

**Comment by magento-engcom-team** [comment]
@kevin-carroll-cl, thank you for your report.
The issue is already fixed in 2.1.8
